src/mechanics/elements/SpectralShapeFunctions.h

```diff
@@ -84,18 +90,18 @@ std::vector<double> ShapeFunctionsTrussLagrange(const double x, const std::vecto
 //! @param x local coordinate where shapes are evaluated
 //! @param nodes local node coordinates
 //! @return shape derivatives evaluated at x.
-std::vector<double> DerivativeShapeFunctionsTrussLagrange(const double x, const std::vector<double>& nodes)
+Eigen::VectorXd DerivativeShapeFunctionsTrussLagrange(const double x, const Eigen::VectorXd& nodes)
 {
-    std::vector<double> result(nodes.size(), 0.);
-    std::vector<double> w = BarycentricWeights(nodes);
-    for (size_t j = 0; j < nodes.size(); j++)
+    Eigen::VectorXd result = Eigen::VectorXd::Zero(nodes.size());
+    Eigen::VectorXd w = BarycentricWeights(nodes);
+    for (int j = 0; j < nodes.size(); j++)
     {
-        for (size_t k = 0; k < nodes.size(); k++)
+        for (int k = 0; k < nodes.size(); k++)
```

Comment:
One (hopefully) last request. Eigen::size() returns something like the size in memory. For a vector, this is equal to the number of its rows. Still, I find it _more_ correct, to use `.rows()` here and in the other places. This leaves no room for doubts.